packages/blaze/lookup.js

```diff
@@ -1,3 +1,6 @@
+/* global Blaze */
+/* eslint-disable import/no-unresolved, no-cond-assign, no-global-assign, prefer-rest-params, no-param-reassign, no-multi-assign */
```

Comment:
Can you please try to not just silence eslint and find better alternatives for at least `no-cond-assign`, `prefer-rest-params`, `no-param-reassign` and `no-multi-assign`?

It's okay to disable `import/no-unresolved` as they are caused by Meteor's package system and we can't work around. At this point we also can't break the hole package by removing global assignments. But the other warnings are usually solveable using ES6.

Reply:
Let me know if you need help on a specific situation here

Reply:
I did a code review to that effect. We have a strong impact by disabling no-param-reassign. Many of the functions do change the value of attributes in objects. 

```
  static _destroy(m, _skipNodes) {
    if (m instanceof DOMRange) {
      if (m.view) Blaze._destroyView(m.view, _skipNodes);
    } else if ((!_skipNodes) && m.nodeType === 1) {
      // DOM Element
      if (m.$blaze_range) {
        Blaze._destroyNode(m);
        m.$blaze_range = null;
      }
    }
  }

```

```
Blaze._createView = function (view, parentView, forExpansion) {
  if (view.isCreated) throw new Error("Can't render the same View twice");

  view.parentView = (parentView || null);
  view.isCreated = true;
  if (forExpansion) view._isCreatedForExpansion = true;

  Blaze._fireCallbacks(view, 'created');
};

```

In some cases we could use the argument default but it does not cover some situations now supported: null, zero. 

```
Blaze._parentData = function (height, _functionWrapped) {
  // If height is null or undefined, we default to 1, the first parent.
  if (height == null) {
    // eslint-disable-next-line no-param-reassign
    height = 1;
  }
```

We should create many local variables to eliminate the warning.

Reply:
We can use nullish coalescing in such situations 

```js
const parent = height ?? 1
```

and if zero is not a valid value then simply or would be sufficient, too

```js
const parent = height || 1
```

I know this may be a bit of a chore but in the long run we will be better with local variables than param reassignment

Reply:
For those cases where object props are updated we should use/add an update fn on the class

```js
m.updateParentView(parentView || null)
```

Reply:
I used no-param-reassign in a few cases and for line. Whenever possible I created a class method to modify attributes. For the remaining cases I used local variables. Let me know if this is fine or if you have any other ideas.

Reply:
I did the push with the changes. 

PS: I have been using Blaze happily since 2015 and have over 500 templates to maintain and bring up to version 3.0 (Meteor/Blaze), I am very concerned about the life of the project :-)